ethers-solc/src/compile/mod.rs

```diff
@@ -218,10 +218,21 @@ impl Solc {
 
     /// Returns the directory in which [svm](https://github.com/roynalnaruto/svm-rs) stores all versions
     ///
-    /// This will be `~/.svm` on unix
+    /// This will be:
+    ///  `~/.svm` on unix, if it exists
+    /// - $XDG_DATA_HOME (~/.local/share/svm) if the svm folder does not exist.
     #[cfg(not(target_arch = "wasm32"))]
     pub fn svm_home() -> Option<PathBuf> {
-        home::home_dir().map(|dir| dir.join(".svm"))
+        match home::home_dir().map(|dir| dir.join(".svm")) {
+            Some(dir) => {
+                if !dir.exists() {
+                    dirs::data_dir().map(|dir| dir.join("svm"))
+                } else {
+                    Some(dir)
+                }
+            }
+            None => dirs::data_dir().map(|dir| dir.join("svm")),
```

Comment:
Isn't this also just SVM_DATA_DIR now?

can we use this instead?

Reply:
Ah, the problem with using this directly is that we'd lock the crate to use svm always instead of conditionally (it's currently a feature)